pysignalscope/scope.py

```diff
@@ -103,61 +137,61 @@ def modify(self, channel_data_factor: Optional[float] = None, channel_data_offse
         :type channel_time_cut_max: float
         :param channel_linestyle: channel linestyle, e.g. '--'
         :type channel_linestyle: str
-        :return: None
-        :rtype: None
+        :return: Scope object
+        :rtype: Scope
         """
         if isinstance(channel_label, str) or channel_label is None:
```

Comment:
Befor assigning any value to the dataset attribute a deep copy is necessary. Otherwise you overwrite the origin data of the data set.
So each function needs a deep copy, like
 cur_channel=copy(channel)
Then the attributes of cur_channel can be updated and cur_channel can be returned

Reply:
fixed